cassovary-core/src/main/scala/com/twitter/cassovary/algorithms/similarity/JaccardSimilarity.scala

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2015 Twitter, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package com.twitter.cassovary.algorithms.similarity
+
+import com.twitter.cassovary.graph.DirectedGraph
+import com.twitter.cassovary.graph.GraphDir.GraphDir
+
+class JaccardSimilarity(val graph: DirectedGraph) extends Similarity {
+
+  override def calculateSimilarity(dir: GraphDir, u: Int, v: Int): Double = {
+    val nodeU = graph.getNodeById(u)
+    val nodeV = graph.getNodeById(v)
+    val neighborsOfU = nodeU.get.neighborIds(dir).toSet
+    val neighborsOfV = nodeV.get.neighborIds(dir).toSet
```

Comment:
It seems like you need a bunch of the same metrics for these similarity calculations, maybe you could move the duplication to a protected method in the Similarity trait?